volatility3/framework/plugins/windows/strings.py

```diff
@@ -81,22 +87,45 @@ def _generator(self) -> Generator[Tuple, None, None]:
         last_prog: float = 0
         line_count: float = 0
         num_strings = len(string_list)
-        for offset, string in string_list:
+        for phys_offset, string in string_list:
             line_count += 1
-            try:
-                revmap_list = [
-                    name + ":" + hex(offset) for (name, offset) in revmap[offset >> 12]
-                ]
-            except (IndexError, KeyError):
-                revmap_list = ["FREE MEMORY"]
-            yield (
-                0,
-                (
-                    str(string, "latin-1"),
-                    format_hints.Hex(offset),
-                    ", ".join(revmap_list),
-                ),
+
+            # We should really take care of this in the revmap generator
+            mapping_entry = revmap.get(
+                phys_offset & 0xFFFFFFFFFFFFF000, [("In Unallocated Space", 0)]
```

Comment:
I'm still keen to know the difference between this and shifting it by 12 (which should be the page bit size not hardcoded to 12, so the previous version wasn't great either)?  Is this a case where we should be canconcializing?  Are the bits above 0xFFFFFFFFFFFFF000 important?

Reply:
Sorry but out of my depth in explaining why I just compared the two versions one works, the other doesn't